include/libtorrent/sha512.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef TORRENT_SHA512_HPP_INCLUDED
+#define TORRENT_SHA512_HPP_INCLUDED
+
+#include "libtorrent/config.hpp"
+#include <cstdint>
+
+namespace libtorrent
+{
+
+	struct sha512_ctx
+	{
+		std::uint64_t length, state[8];
+		size_t curlen;
+		std::uint8_t buf[128];
+	};
+
+	TORRENT_EXTRA_EXPORT int SHA512_init(sha512_ctx* context);
```

Comment:
long term I imagine these implementations should just be the member functions of the respective ``hasher`` class